lib/rubygems/package/tar_reader/entry.rb

```diff
@@ -103,6 +103,13 @@ def file?
   end
 
   ##
+  # Is this tar entry a symlink?
+
+  def symlink?
+    @header.typeflag == "2"
+  end
```

Comment:
The test for this can mirror [`test_directory_eh`](https://github.com/rubygems/rubygems/blob/master/test/rubygems/test_gem_package_tar_reader_entry.rb#L87)